tvm/src/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -304,87 +203,126 @@ void Schedule::to_stage(const Tensor& target,
   }
 }
 
-// stream data between hardware modules
+// Stream data between hardware modules
 void Schedule::stream_to(const Tensor& target, Stage dest, Stage source,
                          Array<Expr> stream_pos, StreamType stream_type,
-                         int channel_depth, std::string new_name) {
+                         int channel_depth, Array<IterVar> axis) {
   Stage target_stage = (*this)[target];
   std::vector<Stage> consumers;
   size_t num_stage = (*this)->stages.size();
-  Buffer target_buffer;
   const ExternOpNode* destOp = dest->op.as<ExternOpNode>();
   const ExternOpNode* srcOp = source->op.as<ExternOpNode>();
 
-  // update kernel def and scope
-  const PlaceholderOpNode* op = target_stage->op.as<PlaceholderOpNode>();
-  bool is_placeholder = op ? true : false;
-  if (is_placeholder) {
-    for (size_t i = 0; i < num_stage; i++) {
-      Stage s = (*this)->stages[i];
-      if (const ExternOpNode* op = s->op.as<ExternOpNode>()) {
-        for (size_t j = 0; j < op->inputs.size(); j++) {
-          if (target == op->inputs[j]) {
-            target_buffer = op->input_placeholders[j];
-            consumers.push_back(s);
-          }
-        }
-      }
-    }
-  } else {  // mark device scope of consumers & update kernel stmts
-    const ExternOpNode* op = target_stage->op.as<ExternOpNode>();
-    target_buffer = op->output_placeholders[0];
-    consumers.push_back(target_stage);
-    for (size_t i = 0; i < num_stage; i++) {
-      Stage s = (*this)->stages[i];
-      if (const ExternOpNode* op = s->op.as<ExternOpNode>()) {
-        for (size_t j = 0; j < op->inputs.size(); j++) {
-          if (target_buffer == op->input_placeholders[j]) {
-            consumers.push_back(s);  // mark buffer in calls
-          }
+  // Extract target buffer and consumers of the channel
+  // When a global buffer is streamed between modules,
+  // it can and only can have two consumers
```

Comment:
Why two consumers? Not understand.

Reply:
Updated the terms here. I meant there should be single producer and single consumer